Code/Framework/AzFramework/AzFramework/Archive/INestedArchive.h

```diff
@@ -71,6 +71,10 @@ namespace AZ::IO
             // multiple times
             FLAGS_DONT_COMPACT = 1 << 5,
 
+            // if this is set, validate header data when opening the archive and validate CRCs when decompressing
+            // & reading files.
+            FLAGS_FULL_VALIDATE = 1 << 10,
```

Comment:
I think the validation of the header data should be separated performing a CRC32 operation of the decompressed data.

The CRC32 calculation over the decompressed data is more extensive than validating the zip local file header

Reply:
So two flags: `FLAGS_VALIDATE_HEADERS` and `FLAGS_FULL_VALIDATE` -- Full validate will add-in the CRC checks?

Reply:
I think they should just be complete separate.
`FLAGS_VALIDATE_LOCALFILEHEADERS` and `FLAGS_VALIDATE_UNCOMPRESSED_CRCS`.